pkgcore/operations/format.py

```diff
@@ -57,7 +58,9 @@ def fetch_one(self, fetchable, observer):
     def failed_fetch(self, fetchable, observer):
         # run pkg_nofetch phase for fetch restricted pkgs
         if 'fetch' in self.pkg.restrict:
-            build_ops = self.domain.build_pkg(self.pkg, observer)
```

Comment:
Right, I meant rebase your fix for that on top of my fix for only running pkg_nofetch when an ebuild is fetch restricted. I was never trying to fix this bug with my previous commit.